### PR TITLE
Get rid of declarativeProps step

### PR DESCRIPTION
**Pull request by abayer**
On top of #94 after refactoring this change out of there in the first place. =)

The declarativeProps...well, why is it even a step? We're basically
just using it internally, we can whitelist it, we can avoid the log
spam of "[Pipeline] declarativeProps" showing up a bunch, we can avoid
cluttering the flow graph... Yeah.

cc @reviewbybees esp @rsandell

### CODE REVIEW

**Comment by rsandell** [comment]
ok, holding off reviewing untill #94 is merged and this is rebased.

**Comment by abayer** [comment]
@rsandell Ready for review now.

File: pipeline-model-definition/src/main/java/org/jenkinsci/plugins/pipeline/modeldefinition/agent/DeclarativeDockerUtils.java [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+        return getLabel(null);
+    }
+
+    @Whitelisted
+    public static String getLabel(@Nullable String override) {
+        if (!StringUtils.isBlank(override)) {
+            return override;
+        } else {
+            for (DockerPropertiesProvider provider : DockerPropertiesProvider.all()) {
+                String label = provider.getLabel(currentRun());
```
Comment (rsandell) [inline pipeline-model-definition/src/main/java/org/jenkinsci/plugins/pipeline/modeldefinition/agent/DeclarativeDockerUtils.java]: add `Run current = currentRun()` outside of for loop

Comment (abayer) [inline pipeline-model-definition/src/main/java/org/jenkinsci/plugins/pipeline/modeldefinition/agent/DeclarativeDockerUtils.java]: Fixing!

**Pull request merged**